Source/WebCore/crypto/mac/CryptoKeyECMac.cpp

```diff
@@ -254,49 +300,49 @@ RefPtr<CryptoKeyEC> CryptoKeyEC::platformImportSpki(CryptoAlgorithmIdentifier id
             return nullptr;
-        
     }
-    return create(identifier, curve, CryptoKeyType::Public, PlatformECKeyContainer(ccPublicKey), extractable, usages);
+    return create(identifier, curve, CryptoKeyType::Public, std::unique_ptr<typename std::remove_pointer<CCECCryptorRef>::type, WebCore::CCECCryptorRefDeleter>(ccPublicKey), extractable, usages);
 }
 
 Vector<uint8_t> CryptoKeyEC::platformExportSpki() const
 {
     size_t expectedKeySize = 2 * keySizeInBytes() + 1; // Per Section 2.3.4 of http://www.secg.org/sec1-v2.pdf
     Vector<uint8_t> keyBytes(expectedKeySize);
     size_t keySize = keyBytes.size();
-    if (UNLIKELY(CCECCryptorExportKey(kCCImportKeyBinary, keyBytes.data(), &keySize, ccECKeyPublic, m_platformKey.get()) || keySize != expectedKeySize))
-        return { };
-
-    // The following addes SPKI header to a raw EC public key.
-    // Once the underlying crypto library is updated to output SPKI EC Key, we should remove this hack.
-    // <rdar://problem/30987628>
-    const uint8_t* oid;
-    size_t oidSize = getOID(namedCurve(), oid);
-
-    // SEQUENCE + length(1) + OID id-ecPublicKey + OID secp256r1/OID secp384r1/OID secp521r1 + BIT STRING + length(?) + InitialOctet + Key size
-    size_t totalSize = sizeof(IdEcPublicKey) + oidSize + bytesNeededForEncodedLength(keySize + 1) + keySize + 4;
-
     Vector<uint8_t> result;
-    result.reserveInitialCapacity(totalSize + bytesNeededForEncodedLength(totalSize) + 1);
-    result.append(SequenceMark);
-    addEncodedASN1Length(result, totalSize);
-    result.append(SequenceMark);
-    addEncodedASN1Length(result, sizeof(IdEcPublicKey) + oidSize);
-    result.append(IdEcPublicKey, sizeof(IdEcPublicKey));
-    result.append(oid, oidSize);
-    result.append(BitStringMark);
-    addEncodedASN1Length(result, keySize + 1);
-    result.append(InitialOctet);
-    result.append(keyBytes.data(), keyBytes.size());
-
-    return result;
+    auto keyBytesData = keyBytes.data();
+    return WTF::switchOn(m_platformKey, [&](const std::unique_ptr<typename std::remove_pointer<CCECCryptorRef>::type, WebCore::CCECCryptorRefDeleter>& options) mutable->Vector<uint8_t> {
+        if (UNLIKELY(CCECCryptorExportKey(kCCImportKeyBinary, keyBytesData , &keySize, ccECKeyPublic, options.get()) || keySize != expectedKeySize))
+            return { };
+        // The following addes SPKI header to a raw EC public key.
+        // Once the underlying crypto library is updated to output SPKI EC Key, we should remove this hack.
+        // <rdar://problem/30987628>
+        const uint8_t* oid;
+        size_t oidSize = getOID(namedCurve(), oid);
+
+        // SEQUENCE + length(1) + OID id-ecPublicKey + OID secp256r1/OID secp384r1/OID secp521r1 + BIT STRING + length(?) + InitialOctet + Key size
+        size_t totalSize = sizeof(IdEcPublicKey) + oidSize + bytesNeededForEncodedLength(keySize + 1) + keySize + 4;
+        result.reserveInitialCapacity(totalSize + bytesNeededForEncodedLength(totalSize) + 1);
```

Comment:
reserveInitialCapacity is usually interesting with uncheckedAppend if possible.
But we need to be sure of the exact init size.
Since this is preexisting code this is probably fine to keep as is but we could do this as a follow-up

Reply:
Ok, we can keep it then